src/play_main_hooks.c

```diff
@@ -429,6 +439,17 @@ void update_rando(PlayState* play) {
             initItems = true;
         }
 
+        if (gMagicFix) { // REPLACE WITH YAML CHECK
+            if (gSaveContext.save.saveInfo.playerData.magic == 0 && gSaveContext.save.saveInfo.playerData.isMagicAcquired && has_unlocked_magic < 2) {
+                if (gSaveContext.save.saveInfo.playerData.magicLevel == 1) {
```

Comment:
I'm fairly sure that this is unnecessary, since it should be impossible to go straight from no magic to double magic. It doesn't hurt anything either though. So up to you, you can leave this check or you can make it unconditionally set it to MAGIC_NORMAL_METER